### PR TITLE
Testrail: automation type tags parsing

### PR DESCRIPTION
**Pull request by andrey-brovko**

### CODE REVIEW

File: GherkinSyncTool.Synchronizers.TestRail/Content/CaseContentBuilder.cs [hunk 1]
```diff
@@ -196,11 +229,33 @@ private string ConvertToStringReferences(Scenario scenario, IFeatureFile feature
...
+
+        private Dictionary<uint, string> GetTestRailAutomationTypes()
+        {
+            var result = new Dictionary<uint, string>();
+            //Get possible automation types
+            var automationTypes = _testRailCaseFields.CaseFields.FirstOrDefault(field => field.SystemName.Equals("custom_automation_type"));
+            //Parse current tag
+            var testRailAutomationTypes = automationTypes?.JsonFromResponse.ToObject<CustomFieldsModel>().Configs.Select(config => config.Options.Items)
+                .FirstOrDefault();
+            //
```
Comment (SergeyRuzin) [inline GherkinSyncTool.Synchronizers.TestRail/Content/CaseContentBuilder.cs:250]: Missing comment?

File: GherkinSyncTool.Synchronizers.TestRail/Content/CaseContentBuilder.cs [hunk 2]
```diff
@@ -17,10 +18,15 @@ public class CaseContentBuilder
         private readonly TestRailSettings _testRailSettings = ConfigurationManager.GetConfiguration<TestRailConfigs>().TestRailSettings;
         private static readonly Logger Log = LogManager.GetLogger(MethodBase.GetCurrentMethod()?.DeclaringType?.Name);
         private readonly Context _context;
+        private readonly TestRailCaseFields _testRailCaseFields;
 
-        public CaseContentBuilder(Context context)
+        private Dictionary<uint, string> _testRailAutomationTypesDir;
```
Comment (vladimir-s-quantori) [inline GherkinSyncTool.Synchronizers.TestRail/Content/CaseContentBuilder.cs:23]: Maybe turn it to autoproperty?

File: GherkinSyncTool.Synchronizers.TestRail/Content/CaseContentBuilder.cs (hunk 1 above)
Comment (andrey-brovko) [reply GherkinSyncTool.Synchronizers.TestRail/Content/CaseContentBuilder.cs]: Fixed

File: GherkinSyncTool.Synchronizers.TestRail/Utils/TestRailCaseFields.cs [hunk 3]
```diff
@@ -1,60 +1,61 @@
...
+using Newtonsoft.Json;
+using TestRail.Types;
+
+namespace GherkinSyncTool.Synchronizers.TestRail.Utils
+{
+    public class TestRailCaseFields
+    {
+        private readonly TestRailClientWrapper _testRailClientWrapper;
+        private readonly TestRailSettings _testRailSettings = ConfigurationManager.GetConfiguration<TestRailConfigs>().TestRailSettings;
+        private List<CaseField> _caseFields;
```
Comment (vladimir-s-quantori) [inline GherkinSyncTool.Synchronizers.TestRail/Utils/TestRailCaseFields.cs:17]: Same as previous comment

**Review by Georgiy-Petrov-Quantori — APPROVED** [review]

**Review by vladimir-s-quantori — APPROVED** [review]

**Pull request merged**